File: contest/5550_diffuse_bomb/solution.py
class Solution:
    def decrypt(self, code: List[int], k: int) -> List[int]:

        enigma = [0] * len(code)

        if k == 0 : return enigma

        for i in range(len(code)) :
            count = k
            sum = 0

            if k > 0 :
                while(count):
                    index = i + count
                    if index > len(code):
                        index = len(code) - index
                    sum += code[index]
                    count = count - 1

            else :
                while(count):
                    index = i + count
                    if index < 0:
                        index = len(code) + index
                    sum += code[index]
                    count = count + 1

            enigma[i] = sum

        return enigma

File: contest/5550_diffuse_bomb/Solution.cpp
#include <iostream>
#include <vector>

using namespace std;

class Solution {
public:
    vector<int> decrypt(vector<int>& code, int K) {
        
        cin.tie(NULL);
        ios_base::sync_with_stdio(false);
        
        vector<int> enigma(code.size(), 0);
        
        if (K == 0) return enigma;
        
        for (int i=0; i<code.size(); i++) {
            int k = K;
            int sum = 0;
            
            if (k > 0)
            while(k != 0) {
                int index = i+(k--);
                if (index > code.size()) index = code.size() - index;
                sum += code[index];
            }
            
            else
            while(k != 0) {
                int index = i+(k++);
                if (index < 0) index = code.size() + index;
                sum += code[index];
            }
            enigma[i] = sum;
        }
        return enigma;
    }
};